player/player.gd

```diff
@@ -10,30 +10,25 @@ var observed_gravity = Vector3()
 var is_grounded = false
 var free_fly = false
 
+var relx = 0
+var rely = 0
+
 onready var last_mouse_pos = get_viewport().get_mouse_pos()
 onready var camera = get_node("camera")
 
 func _ready():
 	Input.call_deferred("set_mouse_mode", Input.MOUSE_MODE_CAPTURED )
 	set_fixed_process(true)
+	set_process_input(true)
 
 func _fixed_process(delta):
-	# Getting movement of mouse for this current frame
-	# get mouse pos
-	var current_mouse_pos = get_viewport().get_mouse_pos()
-	# make difference with last mouse pos
-	var delta_mouse_pos =  current_mouse_pos - last_mouse_pos
-	# if difference is too big then it's a game beginning artifact, ignore it
-	if( current_mouse_pos.distance_to(last_mouse_pos) > 1000 ):
-		delta_mouse_pos = Vector2(0,0)
-	# memorize current mouse pos
-	last_mouse_pos = current_mouse_pos
-
 	# apply rotation of the player according to mouse horizontal movements
-	rotate_y( delta_mouse_pos.x * angular_speed * delta )
+	rotate_y( relx * angular_speed * delta )
 	# apply rotation of the camera according to mouse vertical movements
-	camera.rotate_x( delta_mouse_pos.y * angular_speed * delta )
-	
+	camera.rotate_x( rely * angular_speed * delta )
+	#reset mouse movement
+	relx=0
+	rely=0
```

Comment:
What if I do more than one `_input` before the next `_fixed_process`? You should either change this here, or change it to `+=` below`

Sidenote: `relx = 0` is more readable than `relx=0`